### PR TITLE
[SCB-1297]Environment parsing failed to fix

### PR DESCRIPTION
**Pull request by wuliuqi666**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install -Pit` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java [hunk 1]
```diff
@@ -197,7 +197,11 @@ private void getProperties(ConfigurableEnvironment environment, PropertySource<?
     if (propertySource instanceof EnumerablePropertySource) {
       EnumerablePropertySource<?> enumerablePropertySource = (EnumerablePropertySource<?>) propertySource;
       for (String propertyName : enumerablePropertySource.getPropertyNames()) {
-        configFromSpringBoot.put(propertyName, environment.getProperty(propertyName, Object.class));
+        try {
+          configFromSpringBoot.put(propertyName, environment.getProperty(propertyName, Object.class));
+        } catch (Exception e) {
+          LOGGER.warn("set up spring property source failed. msg: {}", e.getMessage());
```
Comment (WillemJiang) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java]: It's better to include the stack trace for more information.

**Comment by yhs0092** [comment]
Hello, there should be a switch to determine whether the exception should be caught, according to the description of [SCB-1297](https://issues.apache.org/jira/browse/SCB-1297).
By default, this exception should be thrown out and the boot up process should be aborted. And users can change the corresponding config value to enable ServiceComb-Java-Chassis to catch the exception.

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java [hunk 2]
```diff
@@ -197,11 +197,23 @@ private void getProperties(ConfigurableEnvironment environment, PropertySource<?
     if (propertySource instanceof EnumerablePropertySource) {
       EnumerablePropertySource<?> enumerablePropertySource = (EnumerablePropertySource<?>) propertySource;
       for (String propertyName : enumerablePropertySource.getPropertyNames()) {
-        configFromSpringBoot.put(propertyName, environment.getProperty(propertyName, Object.class));
+        try {
+          configFromSpringBoot.put(propertyName, environment.getProperty(propertyName, Object.class));
+        } catch (Exception e) {
+          if (!getIfIgnoreEnvironment()) {
+            throw new RuntimeException("set up spring property source failed.", e);
```
Comment (liubao68) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java]: I think throw e is better

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java [hunk 3]
```diff
@@ -197,11 +197,23 @@ private void getProperties(ConfigurableEnvironment environment, PropertySource<?
...
+        }
       }
       return;
     }
 
     LOGGER.debug("a none EnumerablePropertySource is ignored, propertySourceName = [{}]", propertySource.getName());
   }
+
+  private boolean getIfIgnoreEnvironment() {
+    return (Boolean) ConfigUtil.createLocalConfig().getProperty("servicecomb.configuration.ignoreResolveFailure");
```
Comment (liubao68) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java]: Better to follow configuration conventions, e.g.  servicecomb.config.ignoreResolveFailure

https://docs.servicecomb.io/java-chassis/zh_CN/config/general-config.html

File: foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java [hunk 4]
```diff
@@ -197,11 +197,23 @@ private void getProperties(ConfigurableEnvironment environment, PropertySource<?
...
+        }
       }
       return;
     }
 
     LOGGER.debug("a none EnumerablePropertySource is ignored, propertySourceName = [{}]", propertySource.getName());
   }
+
+  private boolean getIfIgnoreEnvironment() {
+    return (Boolean) ConfigUtil.createLocalConfig().getProperty("servicecomb.config.ignoreResolveFailure");
```
Comment (HuaiJoe) [inline foundations/foundation-config/src/main/java/org/apache/servicecomb/config/ConfigurationSpringInitializer.java]: NPE would be arised if you haven't set any value for  servicecomb.config.ignoreResolveFailure

**Pull request merged**